### DISCUSSION TITLE
Does spaCy require shuffling data before training?

### DISCUSSION THREAD

**Discussion by kanayer** [explosion/spaCy#10208] [Help: Coding & Implementations] [answered]

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @kannaricci ,

Ideally you want to shuffle your data to ensure that the training batches are more representative of the dataset, and that it's not dependent on some order / index. If you set `max_epochs>=0` during training, the training [Corpus](https://spacy.io/api/corpus) is shuffled automatically every epoch, so you don't need to worry about shuffling it by yourself :)

**Reply by kanayer** [reply]
Thank you very much for your answer!
